src/data.jl

```diff
@@ -907,24 +987,33 @@ function jlconvert(::ReadRepresentation{T,DataTypeODR()}, f::JLDFile,
         end for ref in paramrefs]
     end
 
-    path = String(jlconvert(ReadRepresentation{UInt8,Vlen{UInt8}}(), f, ptr, NULL_REFERENCE))
-    parts = split(path, '.')
-    sym = Symbol(parts[1])
-    m = filter(m->nameof(m) == sym, keys(Base.module_keys))
-    if isempty(m)
-        return hasparams ? UnknownType(path, params) : UnknownType(path)
+    mypath = String(jlconvert(ReadRepresentation{UInt8,Vlen{UInt8}}(), f, ptr, NULL_REFERENCE))
+    if hasparams
+        m_3 = _resolve_type_allmodules(rr,
+                                       f,
+                                       ptr,
+                                       header_offset,
+                                       mypath,
+                                       hasparams,
+                                       params)
+    else
+        m_3 = _resolve_type_allmodules(rr,
+                                       f,
+                                       ptr,
+                                       header_offset,
+                                       mypath,
+                                       hasparams,
+                                       nothing)
     end
-    m = pop!(m)
-    for part in parts[2:end]
-        sym = Symbol(part)
-        if !isdefined(m, sym)
-            return hasparams ? UnknownType(path, params) : UnknownType(path)
-        end
-        m = getfield(m, sym)
+    n = length(m_3)
+    m_3_is_not_unknown_type = BitVector(undef, n)
+    for i = 1:n
+        m_3_is_not_unknown_type[i] = _is_not_unknown_type(m_3[i])
     end
-    if !isa(m, DataType) && !isa(m, UnionAll)
-        return hasparams ? UnknownType(path, params) : UnknownType(path)
+    if isempty(m_3_is_not_unknown_type) || !any(m_3_is_not_unknown_type)
+        return hasparams ? UnknownType(mypath, params) : UnknownType(mypath)
     end
+    m = m_3[findfirst(m_3_is_not_unknown_type)]
```

Comment:
Instead of all the array storage and manipulation, could one just use
```julia
for mod in Iterators.flatten((keys(Base.module_keys), stdlibmodules))
    # if mod works, return whatever this function needs to return, otherwise try the next mod
end
```

Reply:
Yeah let me try that.

Reply:
This is resolved now.

> Instead of all the array storage and manipulation, could one just use

I have implemented exactly this suggestion.